src/tests/Thumbnail.test.tsx

```diff
@@ -35,4 +39,29 @@ describe('Thumbnail', () => {
     // Bookmark button
     expect(screen.getByRole('button')).toBeInTheDocument();
   });
+
+  it('should redirect to movie view page on click', async () => {
+    render(
+      <MemoryRouter>
+        <BookmarkProvider>
+          <Routes>
+            <Route index element={<CategoryPage />} />
+            <Route path='movie/:id' element={<MovieView />} />
+          </Routes>
+        </BookmarkProvider>
+      </MemoryRouter>
+    );
+    const user = userEvent.setup();
+
+    // Find the thumbnail of the movie
+    const thumbnails = screen.getAllByRole('movie');
+    expect(thumbnails[0]).toBeInTheDocument();
+
+    // Making sure it found the expected movie
+    expect(thumbnails[0]).toHaveTextContent('The Shawshank Redemption');
+    await user.click(thumbnails[0]);
+
+    const actorHeading = await screen.findByText('The Shawshank Redemption');
```

Comment:
Maybe you can consider using a different variable name, such as 'titleHeading', to enhance clarity? :)

Reply:
Will do!